drivers/ethernet/eth.h

```diff
@@ -42,4 +43,37 @@ static inline void gen_random_mac(uint8_t *mac_addr, uint8_t b0, uint8_t b1, uin
 	sys_rand_get(&mac_addr[3], 3U);
 }
 
+static inline int net_eth_mac_load(const struct net_eth_mac_config *cfg, uint8_t *mac_addr)
+{
+	if (cfg == NULL || cfg->type == NET_ETH_MAC_DEFAULT) {
+		return -ENODATA;
+	}
+
+	/* Copy the static part */
+	memcpy(mac_addr, cfg->addr, cfg->addr_len);
+
+	if (cfg->type == NET_ETH_MAC_STATIC && cfg->addr_len == NET_ETH_ADDR_LEN) {
+		/* Complete static address */
+		return 0;
+	}
```

Comment:
I'd like to have no difference between the static mac address and the default. as most drivers set the mac address from the dt as the initial value of the mac address in the data struct

Reply:
most instance based drivers use something like:
https://github.com/zephyrproject-rtos/zephyr/blob/491498ab9e301b24a3b9ced9234343fb18c87be6/drivers/ethernet/eth_litex_liteeth.c#L333-L335

saving that also in the config struct seems not nessesary.

Reply:
Yes, I'd refactor those to use the `struct net_eth_mac_config` instead.